### PR TITLE
Add test cases for endpoints and fix stuff

### PR DESCRIPTION
**Pull request by yunkee-lee**
A new test case covers making POST requests. I fixed some bugs that were found by the test case.

### CODE REVIEW

**Comment by yunkee-lee** [comment]
ping

File: src/mds/tests/rolling_test_base.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+from mds.mrs.models import BinaryResource, Client, SavedProcedure
+
+import os
+from shutil import rmtree
+import uuid
+
+class RollingTestCase(TestCase):
+    def setUp(self):
+        self.block_size = 1024
```
Comment (SmallFundamentals) [inline src/mds/tests/rolling_test_base.py]: nit: we can define this only once, as opposed to for every test(), unless you anticipate a test changing this

File: src/mds/tests/rolling_test_base.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        if not os.path.exists(self.path):
+            os.makedirs(self.path)
+
+        open(br.data.path, 'w').close()
+        br.save()
+
+    def tearDown(self):
+        rmtree(self.path)
+
+    def _assert_single_chunk_in_binary(self, chunk_index, byte_data):
```
Comment (SmallFundamentals) [inline src/mds/tests/rolling_test_base.py]: camelCase

maybe just assertChunkData

**Comment by SmallFundamentals** [comment]
LGTM, minor comments

**Pull request merged**